scripts/check-host-zero-nanollm_test.sh

```
@@ -73,6 +73,72 @@ rm -f "${tmp_manifest}"
+assert_rc "render_path: nanollm is caught"                   0 render_path_violation_quiet \
+    "${healthy_deps}"$'\n'"${NANOLLM_PREFIX}/go"
+assert_rc "render_path: profileoracle is caught"             0 render_path_violation_quiet \
+    "${healthy_deps}"$'\n'"github.com/invakid404/baml-rest/internal/bamlprofile/profileoracle"
+
+# --- a grep EXECUTION error must fail BOTH halves of check 5 -----------------
+# graph_has_ref is the one helper used for REQUIRED references too, so it cannot
+# collapse "errored" into "present" the way the forbidden-only helpers do: that
+# would make a broken scan look like the required edge is there. Shadow grep with
+# a function that always exits 2 (a real grep error status) and assert the
+# distinct rc 2 propagates into a gate failure on BOTH loops.
+grep() { return 2; }
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)